nibabel/arrayproxy.py

```diff
@@ -360,10 +360,13 @@ def _get_scaled(self, dtype, slicer):
         scl_slope = np.asanyarray(self._slope)
         scl_inter = np.asanyarray(self._inter)
         use_dtype = scl_slope.dtype if dtype is None else dtype
-        slope = scl_slope.astype(use_dtype)
-        inter = scl_inter.astype(use_dtype)
+
+        if np.can_cast(scl_slope, use_dtype):
+            scl_slope = scl_slope.astype(use_dtype)
+        if np.can_cast(scl_inter, use_dtype):
+            scl_inter = scl_inter.astype(use_dtype)
```

Comment:
Unconditionally casting to `int` types would truncate the new parameters.

Reply:
So if it can't cast, but the user asks to, should this maybe spit a warning out?

Reply:
It's a helper method intended to do a best-effort downcast, but not to the detriment of accuracy. If you ask me to return `int`s but it would change the values (beyond a floating point rounding error), then the expected behavior is to keep floats. Spitting out a warning here would be extremely noisy.